query-service-impl/src/main/java/org/hypertrace/core/query/service/prometheus/QueryRequestToPromqlConverter.java

```diff
@@ -0,0 +1,259 @@
+package org.hypertrace.core.query.service.prometheus;
+
+import static org.hypertrace.core.query.service.QueryRequestUtil.getLogicalColumnNameForSimpleColumnExpression;
+
+import com.google.common.collect.Lists;
+import com.google.protobuf.ByteString;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.apache.commons.codec.binary.Hex;
+import org.hypertrace.core.query.service.ExecutionContext;
+import org.hypertrace.core.query.service.QueryRequestUtil;
+import org.hypertrace.core.query.service.QueryTimeRange;
+import org.hypertrace.core.query.service.api.Expression;
+import org.hypertrace.core.query.service.api.Expression.ValueCase;
+import org.hypertrace.core.query.service.api.Filter;
+import org.hypertrace.core.query.service.api.LiteralConstant;
+import org.hypertrace.core.query.service.api.Operator;
+import org.hypertrace.core.query.service.api.QueryRequest;
+import org.hypertrace.core.query.service.api.Value;
+import org.hypertrace.core.query.service.prometheus.PrometheusViewDefinition.MetricConfig;
+
+class QueryRequestToPromqlConverter {
+
+  private final PrometheusViewDefinition prometheusViewDefinition;
+  private final PrometheusFunctionConverter prometheusFunctionConverter;
+
+  QueryRequestToPromqlConverter(PrometheusViewDefinition prometheusViewDefinition) {
+    this.prometheusViewDefinition = prometheusViewDefinition;
+    this.prometheusFunctionConverter = new PrometheusFunctionConverter();
+  }
+
+  PromQLInstantQuery convertToPromqlInstantQuery(
+      ExecutionContext executionContext,
+      QueryRequest request,
+      LinkedHashSet<Expression> allSelections) {
+
+    QueryTimeRange queryTimeRange =
+        executionContext
+            .getQueryTimeRange()
+            .orElseThrow(() -> new RuntimeException("Time Range missing in query"));
```

Comment:
Can we move this to `buildPromqlQueries` to avoid duplication?

Reply:
Actually I need some params from queryTimeRange to build the return object

Reply:
Then can we abstract this in a separate function ?